# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/RemoteOuterWindowProxy.cpp
# Commit: 108d5fb4418e
# Full Hash: 108d5fb4418e78e4dca0a746863e5784c6d80371
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-23 09:52:48
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r d8dea951a032 -r 108d5fb4418e dom/base/RemoteOuterWindowProxy.cpp
--- a/dom/base/RemoteOuterWindowProxy.cpp	Wed Apr 22 15:48:20 2020 +0000
+++ b/dom/base/RemoteOuterWindowProxy.cpp	Wed Apr 22 15:48:17 2020 +0000
@@ -95,7 +95,7 @@
   BrowsingContext* bc = GetBrowsingContext(aProxy);
   uint32_t index = GetArrayIndexFromId(aId);
   if (IsArrayIndex(index)) {
-    const BrowsingContext::Children& children = bc->GetChildren();
+    Span<RefPtr<BrowsingContext>> children = bc->Children();
     if (index < children.Length()) {
       return WrapResult(aCx, aProxy, children[index],
                         JSPROP_READONLY | JSPROP_ENUMERATE, aDesc);
@@ -120,7 +120,7 @@
       return false;
     }
 
-    for (BrowsingContext* child : bc->GetChildren()) {
+    for (BrowsingContext* child : bc->Children()) {
       if (child->NameEquals(str)) {
         return WrapResult(aCx, aProxy, child, JSPROP_READONLY, aDesc);
       }
@@ -132,7 +132,7 @@
 
 bool AppendIndexedPropertyNames(JSContext* aCx, BrowsingContext* aContext,
                                 JS::MutableHandleVector<jsid> aIndexedProps) {
-  int32_t length = aContext->GetChildren().Length();
+  int32_t length = aContext->Children().Length();
   if (!aIndexedProps.reserve(aIndexedProps.length() + length)) {
     return false;
   }